typespec/Applications.Core/secretstores.tsp

```diff
@@ -68,6 +68,15 @@ enum SecretStoreDataType {
 
   @doc("Certificate secret data type")
   certificate,
+
+  @doc("basicAuthentication type is used to represent username and password based authentication and the secretstore resource is expected to have the keys 'username' and 'password'.")
+  basicAuthentication,
+
+  @doc("azureWorkloadIdentity type is used to represent registry authentication using azure federated identity and the secretstore resource is expected to have the keys 'clientId' and 'tenantId'.")
+  azureWorkloadIdentity,
+
+  @doc("awsIRSA type is used to represent registry authentication using AWS IRSA(IAM Roles for Service accounts) and the secretstore resource is expected to have the keys 'roleARN'.")
```

Comment:
Same as above.

Couple of nits (can't add suggestions since the PR is merged):
- Missing space between `IRSA` and `(`
- `key` instead of `keys`